pkg/observability/observability.go

```diff
@@ -151,3 +153,21 @@ func WithBuildInfo(octx context.Context) context.Context {
 
 	return ctx
 }
+
+// RecordLatency calculate and record the latency.
+// Usage example:
+// func foo() {
+// 	 defer RecordLatency(ctx, metric, tag1, tag2)()
+//   // remaining of the function body.
+// }
+func RecordLatency(ctx context.Context, m *stats.Float64Measure, mutators ...*tag.Mutator) func() {
+	start := time.Now()
```

Comment:
Can we make start a parameter to the function to avoid the closure?

```go
defer RecordLatency(ctx, time.Now(), metric, tag1, tag2)
```

Reply:
Done